### PR TITLE
docs(samples): Adding code samples for log reading

### PR DESCRIPTION
**Pull request by m-strzelczyk**
Preparing code samples for [this page](https://cloud.google.com/batch/docs/analyze-job-using-logs).

### CODE REVIEW

File: samples/requirements-test.txt [hunk 1]
```diff
@@ -1,4 +1,5 @@
 pytest==7.1.3
+nox==2022.8.7
```
Comment (kurtisvg) [inline samples/requirements-test.txt:2]: I don't think nox is usually added  in the requirements-test

File: samples/snippets/logs/read_job_logs.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+from google.cloud import batch_v1
+from google.cloud import logging
+
+
+def get_job_logs(project_id: str, job: batch_v1.Job) -> Generator[logging.TextEntry, None, None]:
+    log_client = logging.Client(project=project_id)
+    logger = log_client.logger("batch_task_logs")
+
+    yield from logger.list_entries(filter_=f"labels.job_uid={job.uid}")
```
Comment (kurtisvg) [inline samples/snippets/logs/read_job_logs.py:27]: Does this really need to be a seperate function? Can we not just iterate over `logger.list_entries` directly? 

If not, can we provide some context why, and maybe use an inline function so the client and log can be part of the main function?

File: samples/snippets/logs/read_job_logs.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+# [START batch_job_logs]
+from typing import Generator, NoReturn
+
+from google.cloud import batch_v1
+from google.cloud import logging
+
+
+def get_job_logs(project_id: str, job: batch_v1.Job) -> Generator[logging.TextEntry, None, None]:
+    log_client = logging.Client(project=project_id)
```
Comment (kurtisvg) [inline samples/snippets/logs/read_job_logs.py:24]: Please include this comment for client initialization: https://googlecloudplatform.github.io/samples-style-guide/#clients

**Review by kurtisvg — CHANGES_REQUESTED** [review]

File: samples/snippets/logs/read_job_logs.py (hunk 2 above)
Comment (m-strzelczyk) [reply samples/snippets/logs/read_job_logs.py:27]: I put it in separate function for easier testing. I use this in test, while the printing function can be untested.

I can change it to read the stdout and see if it works that way if you prefer.

**Comment by m-strzelczyk** [comment]
@kurtisvg I applied some of your comments and suggestions. I kept the log retrieving as 2 separate function for easy testing and so that people can easily copy the log retrieving part for their code. Please let me know if you can accept this division.

File: samples/snippets/logs/read_job_logs.py (hunk 2 above)
Comment (kurtisvg) [reply samples/snippets/logs/read_job_logs.py]: It's an explicit goal of the samples style guide to test the output of the function itself and not use return types: https://googlecloudplatform.github.io/samples-style-guide/#result

Comment (m-strzelczyk) [reply samples/snippets/logs/read_job_logs.py]: @kurtisvg I made the sample fit in one function and just print the logs.

**Review by kurtisvg — APPROVED** [review]

**Pull request merged**